composer/utils/data.py

```diff
@@ -95,26 +96,61 @@ def pil_image_collate(batch: List[Tuple[Image.Image, Union[Image.Image, Tensor]]
     return image_tensor, target_tensor
 
 
-def add_dataset_transform(dataset, transform):
+def add_dataset_transform(dataset, transform, location="end", pre_post: str = "pre") -> Dataset:
     """Flexibly add a transform to the dataset's collection of transforms.
 
     Args:
         dataset: A torchvision-like dataset
         transform: Function to be added to the dataset's collection of transforms
+        location (str, torchvision transform), optional: Where to insert the transform in
+            the sequence of transforms. "end" will append to the end or "start" will
+            insert at index 0, a transform (e.g. torchvision.transforms.ToTensor) will
+            insert pre- or post- the first instance of `location`, determined by the
+            `pre_post` argument. Default = "end". If `location` is not in list of
+            transforms, will append to end.
```

Comment:
Would it be safer to error out if `location` is not in the list?

Reply:
Not every dataset has the same list of transforms, nor do they all have `ToTensor()` in the list of transforms. An error would require the user to provide `location` as a hyperparameter for the algorithm, which in turn would require them to know which transforms are in their dataset. This seems like a bit much. Perhaps a `log.warning()` displaying the list of transforms before the function returns?

Reply:
i'm ok with a warning, would prefer an error to avoid silent fail (e.g. was supposed to run before Normalize, but since no Normalize, it's now silently running at the end).

Reply:
I find this API a bit confusing. Thoughts about replace the `location` and `pre_post` with two lists of transforms?

1. `pre_transformtions: Transformations`: Any transformations that, if present on `dataloader.transform`, will come before this new transformation
2. `post_transformations: Transformations`: Any transformations that, if present on `dataloader.transform`, will come this new transformation

If this is not possible (i.e. a post_transform already appears before a pre_transform), then a ValueError would be raised.

Reply:
Alternatively, I think we may be over engineering this. Thoughts about a singular `is_tensor_transform: bool` argument? If it is True, then it should be inserted after ToTensor; if it is False, then it should be inserted before ToTensor. If there is no `ToTensor` present, then it's inserted at the end.

Do our algorithms depend on a specific ordering (other than for the tensors)?

Reply:
I'm also OK with `before=ToTensor()` and/or `after=end` but maybe that makes the input validation a bit difficult. @growlix you have the design decision on the API, so make a decision based on these inputs and we'll go with it!